New tests

LLetal opened the pull request:
Tests of fundamental functions.

LLetal committed:
test_commit

LLetal committed:
Basic_tests

LLetal committed:
new_tests_update

LLetal committed:
test_01

LLetal committed:
Update runtests.jl

LLetal committed:
Mean test maxwell

LLetal committed:
Merge branch 'new_tests' of https://github.com/LLetal/MonteCarloCollisions.jl into new_tests

rouckas commented on test/runtests.jl:
```diff
@@ -19,28 +19,36 @@ end
         @test fluor_test_particle.list[1].v != scatter_v
 
-        @test energy(sodium_test_particle) > energy(fluor_test_particle) > energy(oxygen_test_particle)
+        @test MonteCarloCollisions.energy(sodium_test_particle) > MonteCarloCollisions.energy(fluor_test_particle) > MonteCarloCollisions.energy(oxygen_test_particle)
```
Neni mi jasny vyznam tohoto testu: sodium/fluor/oxygen_test_particle atd... jsou pary nahodne vygenerovanych termalnich castic, takze pro jejich stredni energie sice nejaka (jina) nerovnost plati, ale pro jednotlive castice to klidne muze byt nahodou obracene. Pokud byste vygeneroval mnohem vice castic nez 2, tak by mohlo fungovat:
```suggestion
        @test sum(MonteCarloCollisions.energy(oxygen_test_particle)) > sum(MonteCarloCollisions.energy(fluor_test_particle)) > sum(MonteCarloCollisions.energy(sodium_test_particle))
```

rouckas commented on test/runtests.jl:
```diff
@@ -35,6 +35,26 @@ end
+    nsampl = 1000000
+    
+    oxygen_test = oxygen_test = Neutrals("Oxygen", 300., 16*amu, 2e19)
+    
```
To dvojite radkovani je tady zbytecne - a na vetsine dalsich mist take

rouckas commented on test/runtests.jl:
```diff
@@ -35,6 +35,26 @@ end
+    
+    oxygen_test = oxygen_test = Neutrals("Oxygen", 300., 16*amu, 2e19)
+    
+    std_sample = [norm(random_maxwell_v(oxygen_test.vthermal)) for x in 1:nsampl]
+    
+    test_sample = [norm(random_maxwell_v(oxygen_test.vthermal)) for x in 1:nsampl]
+    
+    mean_test = [random_maxwell_v(oxygen_test.vthermal) for x in 1:nsampl]
+    
+    @test sqrt(8/pi)*oxygen_test.vthermal - std(std_sample)/100 *5 < mean(test_sample)# < sqrt(8/pi)*oxygen_test.vthermal + std(std_sample)/100
```
Na vypocet smerodatne odchylky a stredni hodnoty muzete pouzit ten stejny sample, tedy staci (netestoval jsem)
```
sample = [norm(random_maxwell_v(oxygen_test.vthermal)) for x in 1:nsampl]
v_mean = mean(sample)
v_mean_std = std(sample)/sqrt(nsampl)
v_mean_theory = sqrt(8/pi)*oxygen_test.vthermal
@test v_mean_theory - v_mean_std *5 < mean(test_sample) < v_mean_theory + v_mean_std *5
```

rouckas commented on test/runtests.jl:
```diff
@@ -35,6 +35,26 @@ end
+    
+    @test sqrt(8/pi)*oxygen_test.vthermal - std(std_sample)/100 *5 < mean(test_sample)# < sqrt(8/pi)*oxygen_test.vthermal + std(std_sample)/100
+    
+    @test mean(test_sample) < sqrt(8/pi)*oxygen_test.vthermal + std(std_sample)/100 *5
```
smazat

rouckas commented on test/runtests.jl:
```diff
@@ -35,6 +35,26 @@ end
+    
+    @test mean(test_sample) < sqrt(8/pi)*oxygen_test.vthermal + std(std_sample)/100 *5
+
+    @test mean(mean_test) < [0,0,0]
```
Pouzil bych neco takoveho (nevim, jestli je spravne argument dims...)
    @test all(mean(sample, dim=1) < std(sample, dim=1)/sqrt(nsampl)*5)

rouckas commented on test/runtests.jl:
```diff
@@ -13,58 +13,28 @@ end
-    @test mean(maxwell_test_list) ≈ 16200.
+        @test energy(sodium_test_particle) > energy(fluor_test_particle) > energy(oxygen_test_particle)
 
+        @test norm(reference_particle.list[1].v) != norm(scatter_v)
```
reference particle neni definovana, smazat radek

rouckas commented on test/runtests.jl:
```diff
@@ -19,23 +19,24 @@ end
 end
 
 @testset "maxwell_tests" begin
+<<<<<<< HEAD
```
To sem asi nepatri - smazat radek

rouckas commented on src/MonteCarloCollisions.jl:
```diff
@@ -336,3 +336,5 @@ end
 end
 
 end
+
```
Neni treba pridavat radky

rouckas reviewed (changes requested):
Nektere testy selhavaji, viz report https://github.com/rouckas/MonteCarloCollisions.jl/pull/6/checks?check_run_id=3656894899

doplnil jsem nejake komentare ke kodu, pred mergnutim je treba je opravit.

LLetal commented:
Dobrý den,
Kód jsem upravil dle vašich komentářů, jen nevím, co s ***@***.***
<https://github.com/test> all(mean(sample, dim=1) < std(sample,
dim=1)/sqrt(nsampl)*5)". Julia mi u toho vyhazuje chybovou hlášku i po
upravení dimenzí z "dim=" na "dims=". Při spouštění v Jupyteru mi ostatní
testy prošly a jsem tedy snad připravený na další pull request.
S pozdravem,
Létal

út 21. 9. 2021 v 0:58 odesílatel Stepan Roucka ***@***.***>
napsal:

> ***@***.**** requested changes on this pull request.
>
> Nektere testy selhavaji, viz report
> https://github.com/rouckas/MonteCarloCollisions.jl/pull/6/checks?check_run_id=3656894899
>
> doplnil jsem nejake komentare ke kodu, pred mergnutim je treba je opravit.
> ------------------------------
>
> In test/runtests.jl
> <https://github.com/rouckas/MonteCarloCollisions.jl/pull/6#discussion_r712547672>
> :
>
> >
>
>          scatter_v = MonteCarloCollisions.scatter(sodium_test_particle.list[1].v, fluor_test_particle.list[1].v, sodium_test_particle.m, fluor_test_particle.m, norm(sodium_test_particle.list[1].v - fluor_test_particle.list[1].v), 0)
>
>
>
>          @test fluor_test_particle.list[1].v != scatter_v
>
>
>
> -        @test energy(sodium_test_particle) > energy(fluor_test_particle) > energy(oxygen_test_particle)
>
> +        @test MonteCarloCollisions.energy(sodium_test_particle) > MonteCarloCollisions.energy(fluor_test_particle) > MonteCarloCollisions.energy(oxygen_test_particle)
>
>
> Neni mi jasny vyznam tohoto testu: sodium/fluor/oxygen_test_particle
> atd... jsou pary nahodne vygenerovanych termalnich castic, takze pro jejich
> stredni energie sice nejaka (jina) nerovnost plati, ale pro jednotlive
> castice to klidne muze byt nahodou obracene. Pokud byste vygeneroval mnohem
> vice castic nez 2, tak by mohlo fungovat:
> ⬇️ Suggested change
>
> -        @test MonteCarloCollisions.energy(sodium_test_particle) > MonteCarloCollisions.energy(fluor_test_particle) > MonteCarloCollisions.energy(oxygen_test_particle)
>
> +        @test sum(MonteCarloCollisions.energy(oxygen_test_particle)) > sum(MonteCarloCollisions.energy(fluor_test_particle)) > sum(MonteCarloCollisions.energy(sodium_test_particle))
>
>
> ------------------------------
>
> In test/runtests.jl
> <https://github.com/rouckas/MonteCarloCollisions.jl/pull/6#discussion_r712552369>
> :
>
> > @@ -35,6 +35,26 @@ end
>
>
>
>  end
>
>
>
> ***@***.*** "maxwell_tests" begin
>
> +    nsampl = 1000000
>
> +
>
> +    oxygen_test = oxygen_test = Neutrals("Oxygen", 300., 16*amu, 2e19)
>
> +
>
>
> To dvojite radkovani je tady zbytecne - a na vetsine dalsich mist take
> ------------------------------
>
> In test/runtests.jl
> <https://github.com/rouckas/MonteCarloCollisions.jl/pull/6#discussion_r712558666>
> :
>
> > +    std_sample = [norm(random_maxwell_v(oxygen_test.vthermal)) for x in 1:nsampl]
>
> +
>
> +    test_sample = [norm(random_maxwell_v(oxygen_test.vthermal)) for x in 1:nsampl]
>
> +
>
> +    mean_test = [random_maxwell_v(oxygen_test.vthermal) for x in 1:nsampl]
>
> +
>
> +    @test sqrt(8/pi)*oxygen_test.vthermal - std(std_sample)/100 *5 < mean(test_sample)# < sqrt(8/pi)*oxygen_test.vthermal + std(std_sample)/100
>
>
> Na vypocet smerodatne odchylky a stredni hodnoty muzete pouzit ten stejny
> sample, tedy staci (netestoval jsem)
>
> sample = [norm(random_maxwell_v(oxygen_test.vthermal)) for x in 1:nsampl]
>
> v_mean = mean(sample)
>
> v_mean_std = std(sample)/sqrt(nsampl)
>
> v_mean_theory = sqrt(8/pi)*oxygen_test.vthermal
>
> @test v_mean_theory - v_mean_std *5 < mean(test_sample) < v_mean_theory + v_mean_std *5
>
>
> ------------------------------
>
> In test/runtests.jl
> <https://github.com/rouckas/MonteCarloCollisions.jl/pull/6#discussion_r712562112>
> :
>
> > @@ -35,6 +35,26 @@ end
>
>
>
>  end
>
>
>
> ***@***.*** "maxwell_tests" begin
>
> +    nsampl = 1000000
>
> +
>
> +    oxygen_test = oxygen_test = Neutrals("Oxygen", 300., 16*amu, 2e19)
>
> +
>
> +    std_sample = [norm(random_maxwell_v(oxygen_test.vthermal)) for x in 1:nsampl]
>
> +
>
> +    test_sample = [norm(random_maxwell_v(oxygen_test.vthermal)) for x in 1:nsampl]
>
> +
>
> +    mean_test = [random_maxwell_v(oxygen_test.vthermal) for x in 1:nsampl]
>
> +
>
> +    @test sqrt(8/pi)*oxygen_test.vthermal - std(std_sample)/100 *5 < mean(test_sample)# < sqrt(8/pi)*oxygen_test.vthermal + std(std_sample)/100
>
> +
>
> +    @test mean(test_sample) < sqrt(8/pi)*oxygen_test.vthermal + std(std_sample)/100 *5
>
>
> smazat
> ------------------------------
>
> In test/runtests.jl
> <https://github.com/rouckas/MonteCarloCollisions.jl/pull/6#discussion_r712564864>
> :
>
> > ***@***.*** "maxwell_tests" begin
>
> +    nsampl = 1000000
>
> +
>
> +    oxygen_test = oxygen_test = Neutrals("Oxygen", 300., 16*amu, 2e19)
>
> +
>
> +    std_sample = [norm(random_maxwell_v(oxygen_test.vthermal)) for x in 1:nsampl]
>
> +
>
> +    test_sample = [norm(random_maxwell_v(oxygen_test.vthermal)) for x in 1:nsampl]
>
> +
>
> +    mean_test = [random_maxwell_v(oxygen_test.vthermal) for x in 1:nsampl]
>
> +
>
> +    @test sqrt(8/pi)*oxygen_test.vthermal - std(std_sample)/100 *5 < mean(test_sample)# < sqrt(8/pi)*oxygen_test.vthermal + std(std_sample)/100
>
> +
>
> +    @test mean(test_sample) < sqrt(8/pi)*oxygen_test.vthermal + std(std_sample)/100 *5
>
> +
>
> +    @test mean(mean_test) < [0,0,0]
>
>
> Pouzil bych neco takoveho (nevim, jestli je spravne argument dims...)
> @test <https://github.com/test> all(mean(sample, dim=1) < std(sample,
> dim=1)/sqrt(nsampl)*5)
> ------------------------------
>
> In test/runtests.jl
> <https://github.com/rouckas/MonteCarloCollisions.jl/pull/6#discussion_r712565708>
> :
>
> >
>
> +        @test norm(reference_particle.list[1].v) != norm(scatter_v)
>
>
> reference particle neni definovana, smazat radek
> ------------------------------
>
> In test/runtests.jl
> <https://github.com/rouckas/MonteCarloCollisions.jl/pull/6#discussion_r712567032>
> :
>
> >
>
>          @test norm(reference_particle.list[1].v) != norm(scatter_v)
>
>
>
>  end
>
>
>
>  @testset "maxwell_tests" begin
>
> +<<<<<<< HEAD
>
>
> To sem asi nepatri - smazat radek
> ------------------------------
>
> In src/MonteCarloCollisions.jl
> <https://github.com/rouckas/MonteCarloCollisions.jl/pull/6#discussion_r712567508>
> :
>
> > @@ -336,3 +336,5 @@ end
>
>  end
>
>
>
>  end
>
> +
>
>
> Neni treba pridavat radky
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rouckas/MonteCarloCollisions.jl/pull/6#pullrequestreview-759139073>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AS53TZSAZOM4NNTRTDXT2EDUC64ARANCNFSM5EMCXZ7Q>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

rouckas commented:
Jakou chybu vam to presne vyhazuje? Zkuste toto:
```
all(dropdims(mean(sample, dims=1), dims=1) .< dropdims(std(sample, dims=1), dims=1)./sqrt(nsampl).*5)
```

Nemusite vytvaret novy pull request, melo by stacit, kdy pushnete zmeny do vasi branch new_tests a tady se to automaticky aktualizuje.

LLetal commented:
Dobrý den,
Zkopíroval jsem chybové hlášky, přičemž jsem si všiml, že když vložím do
Julie první část testu samostatně vyhodí to jinou hlášku, než když pouštím
test dohromady:

Pro spouštění dohromady a při spouštění samostatné první části to vyhazuje
tento error:

TypeError: non-boolean (SVector{3, Float64}) used in boolean context

Stacktrace:
 [1] _all
   @ .\reduce.jl:927 [inlined]
 [2] _all
   @ .\reducedim.jl:887 [inlined]
 [3] #all#697
   @ .\reducedim.jl:885 [inlined]
 [4] all(a::Array{SVector{3, Float64}, 0})
   @ Base .\reducedim.jl:885
 [5] top-level scope
   @ In[29]:23
 [6] eval
   @ .\boot.jl:360 [inlined]
 [7] include_string(mapexpr::typeof(REPL.softscope), mod::Module,
code::String, filename::String)
   @ Base .\loading.jl:1094

MethodError: no method matching abs2(::SVector{3, Float64})
Closest candidates are:
  abs2(::Bool) at bool.jl:80
  abs2(::Complex) at complex.jl:265
  abs2(::Missing) at missing.jl:101
  ...


Pokud vložím do kódu samostatně druhou část testu
"dropdims(std(sample, dims=1), dims=1)./sqrt(nsampl).*5" vyhazuje to
tento error:


Stacktrace:
  [1] (::Statistics.var"#12#13")(t::SVector{3, Float64})
    @ Statistics
C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\Statistics\src\Statistics.jl:329
  [2] _reducedim_init(f::Statistics.var"#12#13", op::typeof(+),
fv::typeof(zero), fop::typeof(sum), A::Vector{SVector{3, Float64}},
region::Int64)
    @ Base .\reducedim.jl:117
  [3] reducedim_init(f::Function, op::typeof(+), A::Vector{SVector{3,
Float64}}, region::Int64)
    @ Base .\reducedim.jl:109
  [4] _varm(A::Vector{SVector{3, Float64}}, m::Vector{SVector{3,
Float64}}, corrected::Bool, region::Int64)
    @ Statistics
C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\Statistics\src\Statistics.jl:328
  [5] varm(A::Vector{SVector{3, Float64}}, m::Vector{SVector{3,
Float64}}; corrected::Bool, dims::Int64)
    @ Statistics
C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\Statistics\src\Statistics.jl:326
  [6] _var(A::Vector{SVector{3, Float64}}, corrected::Bool,
mean::Nothing, dims::Int64)
    @ Statistics
C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\Statistics\src\Statistics.jl:368
  [7] var(A::Vector{SVector{3, Float64}}; corrected::Bool,
mean::Nothing, dims::Int64)
    @ Statistics
C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\Statistics\src\Statistics.jl:366
  [8] _std(A::Vector{SVector{3, Float64}}, corrected::Bool,
mean::Nothing, dims::Int64)
    @ Statistics
C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\Statistics\src\Statistics.jl:443
  [9] std(A::Vector{SVector{3, Float64}}; corrected::Bool,
mean::Nothing, dims::Int64)
    @ Statistics
C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\Statistics\src\Statistics.jl:441
 [10] top-level scope
    @ In[32]:1
 [11] eval
    @ .\boot.jl:360 [inlined]
 [12] include_string(mapexpr::typeof(REPL.softscope), mod::Module,
code::String, filename::String)
    @ Base .\loading.jl:1094


Omlouvám se za tu formu, ale nenapadlo mě, kam jinam to šikovněji vložit.
S pozdravem,
Létal

rouckas commented:
Aha, zapomnel jsem, ze pracujeme s Vector{SVector... } a ne s Array. Lze to zprovoznit treba tak, ze vektor konvertujeme funkci reinterpret takto:
```
    vector_sample = [MonteCarloCollisions.random_maxwell_v(oxygen_test.vthermal) for x in 1:nsampl]
    vector_sample_array = reshape((reinterpret(Float64, vector_sample)), (nsampl, 3))
    @test all(dropdims(mean(vector_sample_array, dims=1), dims=1) .< dropdims(std(vector_sample_array, dims=1), dims=1)./sqrt(nsampl).*5)
```

rouckas commented on test/runtests.jl:
```diff
@@ -10,3 +10,70 @@ using Test
+
+    oxygen_test = MonteCarloCollisions.Neutrals("Oxygen", 300., 16*amu, 2e19);
+    
+    oxygen_list = MonteCarloCollisions.load_interactions_lxcat("test_input.txt", electrons, oxygen_test)
```
promenna electrons neni definovana
